app/models/comment.rb

```diff
@@ -344,6 +344,102 @@ def trimmed_content?
     comment.include?(COMMENT_FILTER)
   end
 
+  def self.receive_tweet
+    comments = Comment.where.not(tweet_id: nil)
+    if comments.any?
+      receive_tweet_using_since comments
+    else
+      receive_tweet_without_using_since
+    end
+  end
+
+  def self.receive_tweet_using_since(comments)
+    comment = comments.last
+    since_id = comment.tweet_id
+    tweets = Client.search(ENV["TWEET_SEARCH"], since_id: since_id).collect do |tweet|
+      tweet
+    end
+    tweets.each do |tweet|
+      puts tweet.text
+    end
+    tweets = tweets.reverse
+    check_and_add_tweets tweets
+  end
+
+  def self.receive_tweet_without_using_since
+    tweets = Client.search(ENV["TWEET_SEARCH"]).collect do |tweet|
+      tweet
+    end
+    tweets = tweets.reverse
+    check_and_add_tweets tweets
+    tweets.each do |tweet|
+      puts tweet.text
+    end
+  end
+
+  def self.check_and_add_tweets(tweets)
+    tweets.each do |tweet|
+      if tweet.reply?
+        in_reply_to_tweet_id = tweet.in_reply_to_tweet_id
+        if in_reply_to_tweet_id.class == Fixnum
+          parent_tweet = Client.status(in_reply_to_tweet_id, tweet_mode: "extended")
+          parent_tweet_full_text = parent_tweet.attrs[:text] || parent_tweet.attrs[:full_text]
+          urls = URI.extract(parent_tweet_full_text)
+          node = get_node_from_urls_present(urls)
+          unless node.nil?
+            twitter_user_name = tweet.user.screen_name
+            tweet_email = find_email(twitter_user_name)
+            users = User.where(email: tweet_email)
+            if users.any?
+              user = users.first
+              replied_tweet_text = tweet.text
+              if tweet.truncated?
+                replied_tweet = Client.status(tweet.id, tweet_mode: "extended")
+                replied_tweet_text = replied_tweet.attrs[:text] || replied_tweet.attrs[:full_text]
+              end
+              replied_tweet_text = replied_tweet_text.gsub(/@(\S+)/){|m| "[#{m}](https://twitter.com/#{m})"}
+              replied_tweet_text = replied_tweet_text.gsub('@','')
+              comment = node.add_comment(uid: user.uid, body: replied_tweet_text, comment_via: 2, tweet_id: tweet.id)
+              comment.notify user
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def self.get_node_from_urls_present(urls)
+    urls.each do |url|
+      if url.include? "https://"
+        if url.last == "."
+          url = url[0...url.length-1]
+        end
+        response = Net::HTTP.get_response(URI(url))
+        redirected_url = response['location']
+        if redirected_url != nil && redirected_url.include?(ENV["WEBSITE_HOST_PATTERN"])
+          node_id = redirected_url.split("/")[-1]
+          if !node_id.nil?
+            node = Node.where(nid: node_id.to_i)
+            if node.any?
+              return node.first
+            end
+          end
+        end
+      end
+
+    end
+    return nil
+  end
+
+  def self.find_email(twitter_user_name)
```

Comment:
Hmm, this seems like it could start to push the performance of the system. Are we finding the email address of the twitter user here, and what do we use it for?

Reply:
Meaning fetching all user tags each time -- can we query any other way?

Reply:
We could narrow with: 

```ruby
UserTag.where.not(data: nil)
```

But i bet we could do even better...

Reply:
`UserTag.where('value LIKE (?)', 'oauth:twitter%').where.not(data: nil).count`

shows 145 records currently

Reply:
Apart from this, I think we'll eventually have to figure a way to look up by twitter handle, and to do it more efficiently. The email may not even be the same one, you know?

Reply:
@namangupta01 would you mind opening a new issue for the optimization, and another to try to see if we can look up twitter users by a different system?  Thank you!! And congrats on getting this merged! 🎉

Reply:
Like, maybe we need to store a new column in UserTag that is the user's twitter handle, maybe `handle`? It's not 100% ideal but it will make a difference for optimization as the system grows.

Reply:
And, it'll bypass the issue of the emails not matching. My email is not the same between Twitter and PL.org, actually!